### PR TITLE
Step 2

### PR DESCRIPTION
**Pull request by JohnnyGsy**

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+  "i18n-ally.localesPaths": [
+    "config/locales"
+  ]
+}
```
Comment (kolyanjy) [inline .vscode/settings.json]: to git ignore

File: Gemfile [hunk 2]
```diff
@@ -5,19 +5,19 @@ ruby '2.7.6'
...
 gem 'importmap-rails', '1.0.3'
+gem 'rails-i18n', '~> 7.0.0'
 gem 'sassc-rails', '2.1.2'
 gem 'sprockets-rails', '3.4.2'
-
 # Declaring JSON structures
 gem 'jbuilder', '2.11.5'
-
 # Database
+gem 'jquery-rails'
```
Comment (kolyanjy) [inline Gemfile]: database group 0_o

File: Gemfile [hunk 3]
```diff
@@ -5,19 +5,19 @@ ruby '2.7.6'
...
 gem 'pg', '1.3.5'
-
 # Puma web server
+gem 'popper_js'
```
Comment (kolyanjy) [inline Gemfile]: pls put gems into correct groups

File: Gemfile [hunk 4]
```diff
@@ -36,6 +36,12 @@ gem 'turbo-rails', '1.0.1	'
 # Reduces boot times through caching
 gem 'bootsnap', '1.11.1', require: false
 
+# Installed to prevent the double load of net/protocol
+gem 'net-http'
+
+# Pagination gem
+gem 'pagy', '~> 5.10'
```
Comment (kolyanjy) [inline Gemfile:49]: 👍

File: app/assets/config/manifest.js [hunk 5]
```diff
@@ -1,4 +1,9 @@
 //= link_tree ../images
+//= link application.scss
 //= link_directory ../stylesheets .css
 //= link_tree ../../javascript .js
 //= link_tree ../../../vendor/javascript .js
+//= jquery
+//= jquery_ujs
+//= popper
+//= bootstrap
```
Comment (kolyanjy) [inline app/assets/config/manifest.js]: empty line

Comment (kolyanjy) [inline app/assets/config/manifest.js]: try to move it into `application.js`

File: app/decorators/book_decorator.rb [hunk 6]
```diff
@@ -1,8 +1,13 @@
 class BookDecorator < Draper::Decorator
+  TRUNCATED_LENGTH = { description: 250 }.freeze
```
Comment (kolyanjy) [inline app/decorators/book_decorator.rb]: just value without hash

File: app/assets/config/manifest.js (hunk 5 above)
Comment (JohnnyGsy) [reply app/assets/config/manifest.js]: Already imported to application.js but it didn't work without import into manifest.js

File: app/javascript/controllers/descriprion_controller.js [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+  connect() {
+    this.showLess()
+  }
+
+  showMore() {
+    this.shortTextTarget.hidden = true
+    this.moreButtonTarget.hidden = true
+    this.longTextTarget.hidden = false
+    this.lessButtonTarget.hidden = false
+    console.log('show more')
```
Comment (kolyanjy) [inline app/javascript/controllers/descriprion_controller.js]: remove logger

File: app/javascript/controllers/descriprion_controller.js [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+  }
+
+  showLess() {
+    this.shortTextTarget.hidden = false
+    this.moreButtonTarget.hidden = false
+    this.longTextTarget.hidden = true
+    this.lessButtonTarget.hidden = true
+    console.log('show less')
```
Comment (kolyanjy) [inline app/javascript/controllers/descriprion_controller.js]: the same

File: app/javascript/application.js [hunk 9]
```diff
@@ -1,3 +1,7 @@
 // Configure your import map in config/importmap.rb. Read more: https://github.com/rails/importmap-rails
 import "@hotwired/turbo-rails"
 import "controllers"
+import "jquery"
+import "jquery_ujs"
+import "popper"
+import "bootstrap"
```
Comment (kolyanjy) [inline app/javascript/application.js:7]: empty line

Comment (kolyanjy) [reply app/javascript/application.js:7]: Pls add this rule to rubocop

File: app/assets/config/manifest.js (hunk 5 above)
Comment (kolyanjy) [reply app/assets/config/manifest.js]: okay

File: app/services/sort_books_service.rb [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+  def call
+    find_books
+    order_by_filter
+  end
+
+  private
+
+  def find_books
+    @books = @category_id ? Book.where(category_id: @category_id) : Book.all
+  end
```
Comment (kolyanjy) [inline app/services/sort_books_service.rb]: you can move it to initializer

File: app/services/sort_books_service.rb [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+    created_at_desc: { created_at: :desc },
+    popular_asc: { created_at: :desc },
+    price_asc: { price: :asc },
+    price_desc: { price: :desc },
+    title_asc: { title: :asc },
+    title_desc: { title: :desc }
+  }.freeze
+  def initialize(params)
+    @category_id = params[:category_id]
+    @filter = params[:filter]&.to_sym || BOOK_DEFAULT_FILTER
```
Comment (kolyanjy) [inline app/services/sort_books_service.rb:13]:
```ruby
@filter = params[:filter]&.to_sym || BOOK_FILTERS[:created_at_desc]
```

Comment (kolyanjy) [inline app/services/sort_books_service.rb]:
```suggestion
    @filter = params[:filter]&.to_sym || BOOK_FILTERS[:created_at_desc]
```

File: app/services/sort_books_service.rb [hunk 12]
```diff
@@ -0,0 +1,30 @@
+class SortBooksService
```
Comment (kolyanjy) [inline app/services/sort_books_service.rb]: its a query object

File: config/initializers/assets.rb [hunk 13]
```diff
@@ -2,10 +2,9 @@
 
 # Version of your assets, change this if you want to expire all your assets.
 Rails.application.config.assets.version = '1.0'
-
 # Add additional assets to the asset load path.
 # Rails.application.config.assets.paths << Emoji.images_path
-
+Rails.application.config.assets.precompile += %w[jquery.min.js jquery_ujs.js bootstrap.min.js popper.js]
```
Comment (kolyanjy) [inline config/initializers/assets.rb]: change on manifest.js

File: config/importmap.rb [hunk 14]
```diff
@@ -5,3 +5,7 @@
 pin '@hotwired/stimulus', to: 'stimulus.min.js', preload: true
 pin '@hotwired/stimulus-loading', to: 'stimulus-loading.js', preload: true
 pin_all_from 'app/javascript/controllers', under: 'controllers'
+pin 'jquery', to: 'jquery.min.js', preload: true
+pin 'jquery_ujs', to: 'jquery_ujs.js', preload: true
+pin 'popper', to: 'popper.js', preload: true
+pin 'bootstrap', to: 'bootstrap.min.js', preload: true
```
Comment (kolyanjy) [inline config/importmap.rb:11]: 👍

File: spec/feature/books/show_spec.rb [hunk 15]
```diff
@@ -0,0 +1,19 @@
+RSpec.describe 'books#show', type: :feature do
+  let(:book) { create(:book) }
+
```
Comment (kolyanjy) [inline spec/feature/books/show_spec.rb:3]: pls add specs with increasing count of books or you can add it with specs on cart

File: spec/feature/books/index_spec.rb [hunk 16]
```diff
@@ -0,0 +1,29 @@
+RSpec.describe 'books#index', type: :feature do
+  context 'with books filtering' do
+    let(:category) { create(:category) }
+    let(:another_category) { create(:category) }
+    let(:books_amount) { 2 }
+    let!(:category_books) { BookDecorator.decorate_collection(create_list(:book, books_amount, category: category)) }
+    let!(:another_category_books) do
+      BookDecorator.decorate_collection(create_list(:book, books_amount, category: another_category))
+    end
+
```
Comment (kolyanjy) [inline spec/feature/books/index_spec.rb:10]: add specs for pagination

**Review by kolyanjy — CHANGES_REQUESTED** [review]
Pls fix comment and change base branch to develop

**Pull request closed without merging**